### DISCUSSION TITLE
is there a way to tell jooq to insert the schema in case of a cast to a domain object?

### DISCUSSION THREAD

**Discussion by robstoll** [jOOQ/jOOQ#15842] [Q&A] [answered]
For instance, we have a email domain defined as:
```
create domain myschema.email as citext check ( value ~ '^...$');
```
The domain then looks like:
```
 public static final Domain<String> EMAIL = Internal.createDomain(
          schema()
        , DSL.name("email")
        , org.jooq.impl.DefaultDataType.getDefaultDataType("\"myschema\".\"citext\"")
        , new CitextBinding() // this casts to "citext" and not to myschema.citext
        , Internal.createCheck(null, null, "CHECK ((VALUE OPERATOR(myschema.~) '^..$'::myschema.citext))") // here myschema.citext is used correctly
    );
```
I would like to tell CitextBinding that it shall also include the schema in its cast since otherwise I run into:
```
ERROR: type "citext" does not exist
```
Any idea? (same same with CitextArrayBinding)

**Comment by lukaseder** [comment, marked as answer]
I'm assuming you're referring to `org.jooq.postgres.extensions.bindings.CitextBinding`, which in turn assumes that the `citext` type is on the `search_path`:
https://www.postgresql.org/docs/current/ddl-schemas.html#DDL-SCHEMAS-PATH

So, putting `myschema` on the `search_path` would be one option in your case.

You don't have to use the out of the box `CitextBinding`, you can bring your own, which is schema aware. I don't think there's much value in adding schema mapping to the binding at this point. It seems like too complex a solution for too simple a problem (which people usually solve with the `search_path`).

**Reply by robstoll** [reply]
good with me :+1:
